modules/configuration/src/main/java/org/apache/ignite/internal/configuration/notifications/ConfigurationNotifier.java

```diff
@@ -70,19 +74,23 @@
             DynamicConfiguration<InnerNode, ?> config,
             long storageRevision
     ) {
-        if (oldInnerNode == null || oldInnerNode == newInnerNode) {
+        if (oldInnerNode != null && oldInnerNode == newInnerNode) {
```

Comment:
First part of condition is excessive. `newInnerNode` can't be null

Reply:
You're right.